src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -20452,6 +20452,29 @@ static void Main()
             CompileAndVerify(new[] { source, s_collectionExtensions }, verify: Verification.Skipped, expectedOutput: "[0, 1], ");
         }
 
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/72762")]
```

Comment:
>(Skip = "https://github.com/dotnet/roslyn/issues/72762") [](http://example.com/codeflow?start=13&length=56)

It doesn't look like the test asserts in main now. Consider checking the latest behavior.
```

        [Fact]
        public void RefStruct_04()
        {
            var source = """
                using System.Collections;
                using System.Collections.Generic;

                dynamic d = null;
                S s = [d];

                ref struct S : IEnumerable<int>
                {
                    public IEnumerator<int> GetEnumerator() => throw null;
                    IEnumerator IEnumerable.GetEnumerator() => throw null;
                    public void Add<T>(T t) => throw null;
                }
                """;

            // [https://github.com/dotnet/roslyn/issues/72762](https://github.com/dotnet/roslyn/issues/72762) - There should be diagnostics for S implementing an interface, and
            // there should also be a diagnostic that `s` cannot be used a recevier for a dynamic call.
            CreateCompilation(source).VerifyDiagnostics(
                // (7,16): error CS8343: 'S': ref structs cannot implement interfaces
                // ref struct S : IEnumerable<int>
                Diagnostic(ErrorCode.ERR_RefStructInterfaceImpl, "IEnumerable<int>").WithArguments("S").WithLocation(7, 16)
                );
        }
```